cirq-core/cirq/transformers/gauge_compiling/sqrt_cz_gauge.py

```diff
@@ -14,16 +14,35 @@
 
 """A Gauge transformer for CZ**0.5 gate."""
 
+import numpy as np
+
 from cirq.transformers.gauge_compiling.gauge_compiling import (
     GaugeTransformer,
     GaugeSelector,
     ConstantGauge,
+    Gauge,
 )
 from cirq.ops.common_gates import CZ
 from cirq import ops
 
-SqrtCZGaugeSelector = GaugeSelector(
-    gauges=[ConstantGauge(pre_q0=ops.X, post_q0=ops.X, post_q1=ops.Z**0.5, two_qubit_gate=CZ**-0.5)]
-)
 
-SqrtCZGaugeTransformer = GaugeTransformer(target=CZ**0.5, gauge_selector=SqrtCZGaugeSelector)
+class SqrtCZGauge(Gauge):
+
+    def weight(self) -> float:
+        return 3.0
+
+    def sample(self, gate: ops.Gate, prng: np.random.Generator) -> ConstantGauge:
+        if prng.choice([True, False]):
+            return ConstantGauge(two_qubit_gate=gate)
+        return ConstantGauge(
+            pre_q0=ops.X,
+            post_q0=ops.X,
+            post_q1=ops.S if gate == CZ**0.5 else ops.S**-1,
+            two_qubit_gate=gate**-1,
+            swap_qubits=prng.choice([True, False]),
```

Comment:
I don't think this is working. I tried it out ([colab](https://colab.sandbox.google.com/drive/11Afpe9y2ou9WyUD08WJR9Z_g4s_IxBSo)), and it never seems to switch the qubits.

Reply:
the `_circuit_diagram_info_` method of CZPowGate always puts the exponent on the qubit with heigher index so swapping the qubits doesn't change its circuit representation.

instead print the repr of the new circuits you will get qubits swapped 50% of the time `print(repr(cirq.transformers.SqrtCZGaugeTransformer(circuit)))`

alternativly you could print the qubits of each operation
```py3
for op in cirq.transformers.SqrtCZGaugeTransformer(circuit).all_operations():
  print(op.qubits)
```

Reply:
Sorry, but shouldn't the two `cirq.X` gates be on qubit 0 half of the time and qubit 1 half of the time?

Reply:
that would be a different gauge. what is currently implemented just swaps the order of the qubits of the two qubit gate. I updated it to reverse the entire thing